utils/RudderContext.js

```diff
@@ -34,7 +34,13 @@ class RudderContext {
       screen.width = window.screen.width;
       screen.height = window.screen.height;
       screen.density = window.devicePixelRatio;
-      this.userAgent = navigator.userAgent;
+      // detect brave browser and append to the user agent
+      if (navigator.brave && navigator.brave.isBrave.name === "isBrave") {
```

Comment:
To avoid error in case 'isBrave' gets deprecated/not supported on a different environment, let change to something like below,
`if (navigator.brave && navigator.brave.hasOwnProperty("isBrave")`

Reply:
if (navigator.brave && navigator.brave.hasOwnProperty("isBrave") this won't work as "isBrave" is not a property of the brave object. Instead let's change it to 
`if (navigator.brave && Object.getPrototypeOf(navigator.brave).isBrave)`